checker/match_test.go

```diff
@@ -110,6 +110,33 @@ func TestMatchHTTPMethods(t *testing.T) {
 	}
 }
 
+// HTTP Paths clause with empty list will match any path.
+func TestMatchHTTPPaths(t *testing.T) {
+	testCases := []struct {
+		title   string
+		paths   []*proto.HTTPMatchPathMatchTypes
+		reqPath string
+		result  bool
+	}{
+		{"empty", []*proto.HTTPMatchPathMatchTypes{}, "/foo", true},
+		{"exact", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Exact{Exact: "/foo"}}}, "/foo", true},
+		{"prefix", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Prefix{Prefix: "/foo"}}}, "/foobar", true},
+		{"exact fail", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Exact{Exact: "/foo"}}}, "/joo", false},
+		{"prefix fail", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Prefix{Prefix: "/foo"}}}, "/joobar", false},
+		{"multiple", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Prefix{Prefix: "/joo"}}, {&proto.HTTPMatchPathMatchTypes_Exact{Exact: "/foo"}}}, "/joobar", true},
+		{"exact invalid path", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Exact{Exact: "/foo"}}}, "foo", false},
+		{"exact path with query", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Exact{Exact: "/foo"}}}, "/foo?xyz", true},
+		{"exact path with fragment", []*proto.HTTPMatchPathMatchTypes{{&proto.HTTPMatchPathMatchTypes_Exact{Exact: "/foo"}}}, "/foo#xyz", true},
```

Comment:
I'd like to see the case `prefix: "/foobar"` matched against `/foo?bar` (should be `false`)